src/bootstrap/sanity.rs

```diff
@@ -45,7 +45,7 @@ pub fn check(build: &mut Build) {
             let target = path.join(cmd);
             let mut cmd_alt = cmd.to_os_string();
             cmd_alt.push(".exe");
-            if target.exists() ||
+            if target.is_file() ||
```

Comment:
Are there any reasons behind the mixed use of `is_file()` and `exists()` ?

Also, I don't quite understand what `target.join(cmd_alt)` is for; when we have, say, `cmd=="python"`, `path=="C:\\Python27"`, and thus `cmd_alt=="python\.exe"`, are we not checking for `C:\\Python27\python\python\.exe` which I doubt make sense? I haven't tested it on any Windows PC, though.